components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AuthorizationCodeGrantHandler.java

```diff
@@ -375,55 +377,63 @@ private String buildCacheKeyForToken(String clientId, AuthzCodeDO authzCodeDO) t
     private boolean validateAuthzCodeFromRequest(AuthzCodeDO authzCodeBean, String clientId, String authzCode)
             throws IdentityOAuth2Exception {
 
-        Map<String, Object> params = new HashMap<>();
-        params.put("clientId", clientId);
-        params.put("authorizationCode", authzCode);
-
+        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE,
+                OAuth2Constants.LogConstants.ActionIDs.VALIDATE_AUTHORIZATION_CODE);
+        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+            diagnosticLogBuilder.inputParam(LogConstants.InputKeys.CLIENT_ID, clientId)
+                    .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION);
+        }
         if (authzCodeBean == null) {
             // If no auth code details available, cannot proceed with Authorization code grant
             if (log.isDebugEnabled()) {
                 log.debug("Invalid token request for client id: " + clientId +
                         "and couldn't find persisted data for authorization code: " + authzCode);
             }
             if (LoggerUtils.isDiagnosticLogsEnabled()) {
-                LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, params,
-                        OAuthConstants.LogConstants.FAILED,
-                        "Invalid authorization code received. Couldn't find persisted data for authorization code.",
-                        "validate-authz-code", null);
+                diagnosticLogBuilder.resultMessage("Invalid authorization code received. Couldn't find persisted data" +
+                                " for authorization code.")
+                        .inputParam("authorization code", authzCode)
```

Comment:
don't we have a constant for this?

Reply:
https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2105/files#diff-4463240780e502d868aa2a5515e50e208ab27fc48973069b6f3485b09bc6c2c1R426

Reply:
Fixed with 358594f1d1df31773a93b3f1cf93ac0c8a1b119e